crates/ruff/src/rules/flake8_raise/rules/unnecessary_paren_on_raise_exception.rs

```diff
@@ -52,6 +52,18 @@ pub(crate) fn unnecessary_paren_on_raise_exception(checker: &mut Checker, expr:
         range: _,
     }) = expr
     {
+        let Expr::Name(ast::ExprName { id, .. }) = func.as_ref() else {
+            return;
+        };
```

Comment:
I think we need to allow arbitrary expressions here -- this would false-negative on anything accessed through an attribute:

```py
raise module.Error()
```

Reply:
What would be the best way to go about this, style and performance wise? Pattern matching against all possible expressions or locate the call by range and get the `id` from the string representation of that?

Reply:
I'd probably do like:

```rust
if let Expr::Name(ast::ExprName { id, .. }) = func.as_ref() {
    if let Some(binding_id) = checker.semantic().lookup(id) {
        let binding = checker.semantic().binding(binding_id);
        if binding.kind.is_function_definition() {
            return;
        }
    }
```

Reply:
Ah check. I see what you mean now. That snippet complains about:
```
error[E0596]: cannot borrow data in a `&` reference as mutable
  --> crates/ruff/src/rules/flake8_raise/rules/unnecessary_paren_on_raise_exception.rs:56:39
   |
56 |             if let Some(binding_id) = checker.semantic().lookup(id) {
   |                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot borrow as mutable
```

But I'll fix that and push changes. Thanks!

Reply:
Hmm that's in `fn lookup(&mut self, symbol: &str)` by the looks of it. And it's implementation can't be taken over cause `global()` is private. I'll keep the structure but use the previous logic for now. Let me know if there's a better fix for the above